### ISSUE TITLE
Empty tab as a result of invalid OPML

### ISSUE THREAD

**Issue by PostMonsterG** [scripting/drummerSupport#151]
### Discussion

- I have a file, which, when opened in Drummer, appears as an empty tab. 
- The file, however, is not empty. It contains a valid OPML file, plus four characters
- Drummer OPML files end with`\t</opml>`, that is, the final line of the file consists of a tab and `</opml>`. There is no newline., it ends with `\t</opml>\nl>\n`

**Comment by scripting** [comment]
Is there a reason you can't provide the file itself?

**Comment by PostMonsterG** [comment]
Because it wasn't relevant. I think I misunderstood what drummerSupport is for, you can close and delete this.

**Comment by scripting** [comment]
Of course I can close the issue, but you seem to be reporting a bug. Why did you post this? Do you want me to investigate?

**Comment by scripting** [comment]
I re-read your post and I think what you're doing is trying to provide support not asking for support. If that is the case, that is not what this place is for, and you should delete the post. 

However, I think the best way to deal with this is to teach Drummer not to have a problem with such files. For that I would need an example of how to produce the problem. Of course I can create a file that has invalid OPML and see what happens (and I will do that), or you could help me out and save some time and give me a pointer to the file that caused the problem. That would be an appropriate use of this forum.

Gary, I have so little time, and am so stretched, I don't want these kinds of issues to keep popping up. Please stop trying to test the boundaries. Thanks.

**Comment by PostMonsterG** [comment]
As I said, and you confirmed, I misunderstood the purpose. Your question immediately clarified that for me, and I instantly regretted the intrusion.

Any pushing of boundaries was unintentional. I didn't delete this myself because I didn't realize I can do that sort of thing.

**Comment by scripting** [comment]
Thank you. 'd like to investigate the problem and make Drummer more resilient so the blank tabs don't happen. Are you willing to help?

**Comment by PostMonsterG** [comment]
Yes, of course.

**Comment by scripting** [comment]
I don't have any time right now to dig into this, but I did check the file for validity and it's not valid XML.

It's definitely not empty.

In line 308 there are two _created_ atts. 

Did you edit the file by hand by an chance?

Dave

**Comment by PostMonsterG** [comment]
No, everything is edited in Drummer, using Drummer's UI. Whatever is in that file—I have not looked yet—was written by Drummer.

**Comment by PostMonsterG** [comment]
I pulled my latest backup from GitHub. This version has dateModified Thu, 16 Jun 2022 21:32:34 GMT. The broken file has dateModified Thu, 16 Jun 2022 21:33:22 GMT, less than a minute later. Line 308 looks OK in the backup version, but the broken file looks pretty broken from that point on down. I will be looking into the diffs to see if I can determine a specific string that can trigger this, and thereby construct an appropriate test case that isn't 650K.

[blog-broken.opml.zip](https://github.com/scripting/drummerSupport/files/8930796/blog-broken.opml.zip)
[blog-lastbackup.opml.zip](https://github.com/scripting/drummerSupport/files/8930797/blog-lastbackup.opml.zip)
